### PR TITLE
Mcq image field updated

### PR DESCRIPTION
**Pull request by Novfal-202**
Missed functionality for adding mcq image in cloud and storing reference in respective field.

### CODE REVIEW

**Comment by Novfal-202** [comment]
@HarishGangula could you review these PR .

File: src/controllers/question.ts [hunk 1]
```diff
@@ -497,7 +532,9 @@ const formatQuestionStageData = async (stageData: any[]) => {
         sub_skills: obj?.sub_skill?.map((subSkill: string) => subSkills.find((sub: any) => sub?.name?.en === subSkill)),
         question_body: {
           numbers: { n1: grid_fib_n1, n2: grid_fib_n2 },
-          options: obj?.question_type === 'Mcq' ? [mcq_option_1, mcq_option_2, mcq_option_3, mcq_option_4, mcq_option_5, mcq_option_6] : undefined,
+          question_image: mcq_question_image,
+          options:
+            obj?.question_type === 'Mcq' ? [mcq_option_1, mcq_option_2, mcq_option_3, mcq_option_4, mcq_option_5, mcq_option_6].filter((option) => option !== null && option.trim() !== '') : undefined,
```
Comment (HarishGangula) [inline src/controllers/question.ts:537]: convert to lowercase and compare

File: src/controllers/question.ts [hunk 2]
```diff
@@ -355,7 +355,41 @@ const processQuestionMediaFiles = async () => {
...
+              data: null,
+            },
+          };
+        }
+      }
+      const {
+        question_type,
+        body: { mcq_question_image = null },
+      } = question;
+      if (question_type === 'Mcq' && mcq_question_image) {
```
Comment (HarishGangula) [inline src/controllers/question.ts:371]: convert to lowercase and compare

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/controllers/question.ts (hunk 2 above)
Comment (Novfal-202) [reply src/controllers/question.ts:371]: Currently, for both question type and L1 skill, the first letter is capitalised, followed by lowercase letters. This approach has also been discussed with the content team.

File: src/controllers/question.ts (hunk 1 above)
Comment (Novfal-202) [reply src/controllers/question.ts:537]: Currently, for both question type and L1 skill, the first letter is capitalized, followed by lowercase letters. This approach has also been discussed with the content team.

File: src/controllers/question.ts [hunk 3]
```diff
@@ -497,8 +532,12 @@ const formatQuestionStageData = async (stageData: any[]) => {
         sub_skills: obj?.sub_skill?.map((subSkill: string) => subSkills.find((sub: any) => sub?.name?.en === subSkill)),
         question_body: {
           numbers: { n1: grid_fib_n1, n2: grid_fib_n2 },
-          options: obj?.question_type === 'Mcq' ? [mcq_option_1, mcq_option_2, mcq_option_3, mcq_option_4, mcq_option_5, mcq_option_6] : undefined,
-          correct_option: obj?.question_type === 'Mcq' ? mcq_correct_options : undefined,
+          question_image: mcq_question_image,
+          options:
+            obj?.question_type.toLowerCase() === 'mcq'
```
Comment (HarishGangula) [inline src/controllers/question.ts:537]: add optional chaining for toLowerCase method also

File: src/controllers/question.ts [hunk 4]
```diff
@@ -355,7 +355,41 @@ const processQuestionMediaFiles = async () => {
...
+              data: null,
+            },
+          };
+        }
+      }
+      const {
+        question_type,
+        body: { mcq_question_image = null },
+      } = question;
+      if (question_type.toLowerCase() === 'mcq' && mcq_question_image) {
```
Comment (HarishGangula) [inline src/controllers/question.ts:371]: add optional chaining for toLowerCase method also

**Review by HarishGangula — CHANGES_REQUESTED** [review]

File: src/controllers/question.ts (hunk 3 above)
Comment (Novfal-202) [reply src/controllers/question.ts]: updated.

File: src/controllers/question.ts (hunk 4 above)
Comment (Novfal-202) [reply src/controllers/question.ts]: updated

**Review by HarishGangula — APPROVED** [review]

**Pull request merged**